core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -106,6 +106,8 @@ protected BikeCommonFlagEncoder(int speedBits, double speedFactor, int maxTurnCo
         unpavedSurfaceTags.add("wood");
 
         maxPossibleSpeed = 30;
+        // todo: would like to do it like this but avgSpeedEnc is not created yet
+//        maxPossibleSpeed = avgSpeedEnc.getNextStorableValue(30);
```

Comment:
We set the max possible speed before the `avgSpeedEnc` is created so we have no chance of using `avgSpeedEnc` to find a value that works for `maxPossibleSpeed` (depends on `speedFactor`).

Reply:
And can we instead move the initialization of maxPossibleSpeed after avgSpeedEnc is created (as it is not final)?

Reply:
I think we should not do this. Not initializing avgSpeedEnc in the constructor is bad already. Not initializing the max speed because of this would be even worse. I changed the code so the avgSpeedEnc is initialized in the constructor now.